libqtile/core/state.py

```diff
@@ -42,8 +42,6 @@ def __init__(self, qtile):
         for group in qtile.groups:
             if isinstance(group, ScratchPad):
                 self.scratchpads[group.name] = group.get_state()
-                for dd in group.dropdowns.values():
-                    dd.hide()
```

Comment:
What's the rationale for removing this?

Reply:
These lines were hiding all windows that were visible when the state was saved. Setting the state should be enough here so that windows that were hidden are still hidden and windows that were visible are visible.